hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -114,6 +120,13 @@ public ObjectEndpoint() {
     customizableGetHeaders.add("Content-Encoding");
   }
 
+  @PostConstruct
+  public void init() {
```

Comment:
NIT: It can be slightly better to use

```
  @Inject
  private OzoneConfiguration ozoneConfiguration;
```

And you don't need to expose `getClient()` in the `EndpointBase`. 

(Didn't try, but it should work, IMHO).

Reply:
Thank you for let me know this high technology